pennylane/devices/default_qubit.py

```diff
@@ -582,7 +595,7 @@ def execute(
                 _wrap_simulate,
                 vanilla_circuits,
                 seeds,
-                [self._prng_key] * len(vanilla_circuits),
+                self.get_prng_keys(num=len(vanilla_circuits)),
```

Comment:
The goal of this PR is each call to `simulate` does not uses the same PRNG key. In the current setup however, multithreaded execution will not use the same keys as it branches keys with an n-ary tree while the sequential execution branches with a binary tree, and these do not yield the same key sequences in JAX. But do we want to enforce the same key sequence between sequential and multithreading?

Reply:
This can be achieved with the changes in [8546dc3](https://github.com/PennyLaneAI/pennylane/pull/5515/commits/8546dc3e619fb8103a10182a73d139022fe9ed33).